module/Photo/src/Service/AclService.php

```diff
@@ -12,19 +14,28 @@ protected function createAcl(): void
         $this->acl->addResource('photo');
         $this->acl->addResource('album');
         $this->acl->addResource('tag');
+        $this->acl->addResource('vote');
 
         // Only users and 'the screen' are allowed to view photos (and its details) and albums
         $this->acl->allow('user', 'album', 'view');
-        $this->acl->allow('user', 'photo', ['view', 'vote', 'download', 'view_metadata']);
+        $this->acl->allow('user', 'photo', ['view', 'download', 'view_metadata']);
 
         $this->acl->allow('apiuser', 'photo', 'view');
         $this->acl->allow('apiuser', 'album', 'view');
 
-        // Users are allowed to view, remove and add tags
+        // Users are allowed to view, remove and add tags, and view and add their votes
         $this->acl->allow('user', 'tag', ['view', 'add', 'remove']);
+        $this->acl->allow('user', 'vote', ['view', 'add']);
 
         $this->acl->allow('photo_guest', 'photo', 'view');
         $this->acl->allow('photo_guest', 'album', 'view');
         $this->acl->allow('photo_guest', 'photo', ['download', 'view_metadata']);
+
+        // Graduates may not view photos/albums that were made after their membership ended.
+        $this->acl->deny('graduate', 'album', 'view', new IsAfterMembershipEnded());
+        $this->acl->deny('graduate', 'photo', ['view', 'download', 'view_metadata'], new IsAfterMembershipEnded());
+        // Graduates may not tag people or vote for the photo of the week. This applies to all photos.
+        $this->acl->deny('graduate', 'tag', ['add', 'remove']);
+        $this->acl->deny('graduate', 'vote', 'add');
```

Comment:
I wasn't able to find in [MemberController.php](module/Decision/src/Controller/MemberController.php), [AclService.php](module/Photo/src/Service/AclService.php) or [the view](module/Decision/view/decision/member/view.phtml) something that indicated that viewing member pages is restricted. It does look like searching is no longer possible as that permission is only given to members and not graduates

Reply:
It is in the `Decision` `AclService`:

https://github.com/GEWIS/gewisweb/blob/2e0e8220b7c276bd9db5f37c246ac2bb4fa3db50/module/Decision/src/Service/AclService.php#L44